code/datums/supplypacks.dm

```diff
@@ -55,6 +55,10 @@ GLOBAL_LIST_INIT(all_supply_groups, list(SUPPLY_EMERGENCY,SUPPLY_SECURITY,SUPPLY
 	var/list/announce_beacons = list() // Particular beacons that we'll notify the relevant department when we reach
 	var/special = FALSE //Event/Station Goals/Admin enabled packs
 	var/special_enabled = FALSE
+	/// The number of times one can order a cargo crate, before it becomes restricted. -1 for infinite
+	var/order_limit = -1
```

Comment:
Broader idea here. If the initial value of this is -1 on supplypack/New(), set it to a random value between 800-900, (quite literally, `rand(800, 900)`), an show the stock level on the supply console.

This also opens up opportunities for other crates to be restricted in a nicer fashion.

Reply:
Working on that, currently got it working, however going to try to get the number displayed to update each time an order is approved, rather than having to open and close the menu.